Utils/Nginx/test.cc

```diff
@@ -35,20 +35,19 @@ DEFINE_bool(nginx_overwrite_golden_files, false, "Set to true to have Nginx gold
 
 TEST(Nginx, FullConfig) {
   using current::FileSystem;
-  using namespace current::utils::nginx::config;
+  using namespace current::nginx::config;
 
   FullConfig config;
-  config.AddSimpleDirective("worker_processes", "1");
-  config.AddBlockDirective("events", "").AddSimpleDirective("worker_connections", "1024");
-  config.http.AddSimpleDirective("default_type", "application/octet-stream");
-  config.http.AddSimpleDirective("keepalive_timeout", "100");
-  auto& server = config.http.AddServer(8910);
-  EXPECT_THROW(config.http.AddServer(8910), current::utils::nginx::PortAlreadyUsedException);
-  server.AddSimpleDirective("server_name", "myserver");
-  server.AddDefaultLocation("/").AddSimpleDirective("proxy_pass", "http://127.0.0.1:8888/");
-  auto& complex_location = server.AddLocation("~ /a|/b");
-  complex_location.AddNestedLocation("~ /a");
-  complex_location.AddNestedLocation("~ /b");
+  config.Add(SimpleDirective("worker_processes", "1"));
+  config.Add(BlockDirective("events", {SimpleDirective("worker_connections", "1024")}));
```

Comment:
A test doing `config.Add(BlockDirective("foo").Add(SimpleDirective("bar", "baz")).Add(SimpleDirective("meh", "nah")));` would be nice too.

Great code, LGTM++ && merging.